src/Infrastructure.Dapper/Repositories/CollectionRepository.cs

```diff
@@ -311,7 +315,9 @@ public async Task<ICollection<CollectionAccessSelection>> GetManyUsersByIdAsync(
 
     public class CollectionWithGroupsAndUsers : Collection
     {
-        public DataTable Groups { get; set; }
-        public DataTable Users { get; set; }
+        [DisallowNull]
+        public DataTable? Groups { get; set; }
```

Comment:
Question: It seems a bit surprising to declare the elements as nullable with `?` and then annotate them with `DisallowNull`, what is the difference between this and just marking them non-nullable?

Reply:
This one is tough and can really go multiple ways so happy for some input. My reasoning for this one was because these properties are `null` (albeit very briefly) after the `JsonSerializer.Deserialize<CollectionWithGroupsAndUsers>(JsonSerializer.Serialize(obj))!` calls above. So that is why it feels more honest to say they might be null if someone where to attempt to access it. But then `DisallowNull` makes it so that it will still warn if someone would try to assign null to it. Which is seeming undesirable since we do  this `users != null ? users.ToArrayTVP() : Enumerable.Empty<CollectionAccessSelection>().ToArrayTVP();` whole dance instead of doing `users?.ToArrayTVP()`. 

My bigger preference is to probably not even need this class at all and also skip using json to do a clone-y thing and then there wouldn't be a period of time where the object is in a not good state.

Reply:
Ah that makes sense, the whole deserialize/serialize thing is definitely a bit suboptimal yeah. I agree with you that `DisallowNull` is better than nothing, though.